homework-submission/week5-homework/src/components/TodoItem.js

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import Deadline from "./Deadline";
+class TodoItem extends React.Component {
+  state = {
+    isEditing: false
+  };
+  OnEdit = () => {
+    this.setState({ isEditing: true });
+  };
+  onCancel = () => {
+    this.setState({ isEditing: false });
+  };
+  onUpdateClick = e => {
+    e.preventDefault();
+    let id = this.props._id;
+    let newTodo = this.editedTodo.value;
+    this.props.onUpdate(id, newTodo);
+    this.setState({ isEditing: false });
+  };
+  renderButtons() {
+    if (this.state.isEditing) {
+      return (
+        <td className="actionButton todos-td">
+          <button onClick={this.onUpdateClick}> Update </button>
+          <button onClick={this.onCancel}>Cancel </button>
+        </td>
+      );
+    }
+    return (
+      <td className="actionButton todos-td">
+        <button onClick={this.OnEdit}> Edit </button>
+        <button onClick={this.props.onRemove}>Remove </button>
+      </td>
+    );
+  }
+  renderTodos() {
+    let done = this.props.done;
+    let listStyle = {
+      textDecoration: "line-through"
+    };
+    if (this.state.isEditing) {
+      return (
+        <td className="todos-td">
+          <form onSubmit={this.onUpdateClick}>
+            <input
+              defaultValue={this.props.description}
+              ref={el => (this.editedTodo = el)}
+              className="editing-input"
+            />
+          </form>
+        </td>
+      );
+    }
+    return (
+      <td style={done ? listStyle : {}} className="todos-td">
+        {this.props.children}
+        <strong> {this.props.description} </strong>
+      </td>
+    );
+  }
+  render() {
+    return (
+      <tr>
+        {this.renderTodos()}
+        <td className="todos-td">
+          <Deadline deadline={this.props.deadline} />
+        </td>
+        {this.renderButtons()}
```

Comment:
Here we may have the following while it's same work as the component:
```JSX
<this.renderTodos />
<td className="todos-td">
  <Deadline deadline={this.props.deadline} /> 
</td>
<this.renderButtons />
```
or if you did the advise 😃:
```JSX
<Todos />
<td className="todos-td">
  <Deadline deadline={this.props.deadline} /> 
</td>
<Buttons />
```